### ISSUE TITLE
False positive ``Instance of '_CountingAttr' has no 'append' member`` with attrs

### ISSUE THREAD

**Issue by peendebak** [pylint-dev/pylint#7884]
### Bug description

Pylint is generating warnings on dataclasses defined with [attrs](https://github.com/python-attrs/attrs).
In particular on lists annotated with `field`, e.g. `values: list[str] = field(factory=list)`.

With `a.py` equal to
```python
# Minimal example 

from attrs import define, field

@define
class TestResult:
    """ Test class """

    name: str
    values: list[str] = field(factory=list)
    values2: list[str] = []


    def add_value(self, string : str) -> None:
        """ Ädd value """
        self.values.append(string)

    def hello(self):
        """ Print hello """
        print('hi')
        self.values2.append('hi')
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module qi_analysis.experiments.x
x.py:16:8: E1101: Instance of '_CountingAttr' has no 'append' member (no-member)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 4.00/10, +1.00)
```

### Expected behavior

No pylint errors

### Pylint version

```shell
pylint 2.15.7
astroid 2.12.13
Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)]
```

**Comment by alonme** [comment]
This seems to be originating from the way astroid detects if a class is decorated by attrs [see](https://github.com/PyCQA/astroid/blob/ce07459e1e6cd636dd3c4bb26b7cadeec8de6001/astroid/brain/brain_attrs.py#L18-L31).

This means that it will not detect attrs when using it as is shown in the official [docs](https://www.attrs.org/en/stable/examples.html#basics) .

So - by changing your code to use `attrs.define` as the decorator the problem is solved

```
# Minimal example

import attrs
from attrs import define, field

@attrs.define
class TestResult:
    """ Test class """

    name: str
    values: list[str] = field(factory=list)
    values2: list[str] = []


    def add_value(self, string : str) -> None:
        """ Ädd value """
        self.values.append(string)

    def hello(self):
        """ Print hello """
        print('hi')
        self.values2.append('hi')
```

**Comment by alonme** [comment]
I opened a PR with a POC to improve the way that astroid detects attrs,
However i think that if we don't get that in, we should add some pointer in the docs / warning to tell users to use `@attrs.define` to make it work

**Comment by alonme** [comment]
@Pierre-Sassoulas Which should way should we go here?

1. Add hints for which `attrs` decorators are "valid" to the docs?
2. My Astroid PR?
3. Add flags to enable more `attrs` decorators to be treated?

**Comment by Pierre-Sassoulas** [comment]
I'm not super up to date on this issue, but creating/upgrading an astroid brain is almost always the easy way to fix things, so I would say your MR.
